Islandora/islandora_mirador
# Update mirador

# What does this Pull Request do?

Provide a brief description of what the intended result of the PR will be and/or what
 problem it solves.

* **Related GitHub Issue**: (link)

https://github.com/Islandora/islandora_mirador/issues/4

* **Other Relevant Links**: (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What's new?

Updates Mirador to be able to use either a local copy of the Mirador app or one hosted on a CDN.

Changes the default to a remote Mirador hosted on Github Pages that contains several plugins and is updated to Mirador 3.3.

Adds support to enable individual plugins in the admin interface.

* Does this change add any new dependencies?  No
* Does this change require any other modifications to be made to the repository 
 (i.e. Regeneration activity, etc.)?  No
* Could this change impact execution of existing code? Yes

Updated Mirador so any new behaviours will appear in Islandora.

# How should this be tested?

Test that the update hook runs properly.

Go to the Admin settings page for Islandora Mirador and test the new settings.

To test text overlay integration specifically:

Follow the testing instructions for  https://github.com/Islandora/islandora/pull/897

GO to Admin > Structure > Views and edit the IIIF Manifest view.

In the manifest you want to use, add the hOCR media field you created. Then click the Settings for the IIIF Manifest views plugin, and select that field in the list for hOCR Structured Text.

GO to the settings for Islandora Mirador and enable the Text Overlay and Image Tools plugins.

You may need to clear cache. Go to the node for the Paged Content item you created and observe that the plugins are working.

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/82412/202558620-2bf80d74-5f66-43b7-9360-8f159701793c.png">

# Documentation Status

* Does this change existing behaviour that's currently documented? Updated readme 
* Does this change require new pages or sections of documentation? 
* Who does this need to be documented for?
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
